### ISSUE TITLE
Prompt mode fails to start, after installing 'topydo[prompt]'

### ISSUE THREAD

**Issue by RonCam** [bram85/topydo#192]
I am attempting to set-up the _topydo Prompt mode_ by following these directions, to quote:
## Before first use
Prompt mode requires an additional dependency prompt_toolkit that needs to be installed first:
pip3 install 'topydo[prompt]'

So, I have done this in terminal, expecting to install the dependency (adding --upgrade and --user, as previously suggested):

```
ron@ron-gnulinux:~/.local/bin > pip3 install 'topydo[prompt]' --upgrade --user
Requirement already up-to-date: topydo[prompt] in /home/ron/.local/lib/python3.4/site-packages
  Installing extra requirements: 'prompt'
Cleaning up...
```

But, when I open this directory:
/home/ron/.local/lib/python3.4/site-packages/
I don't see a listing for _prompt_toolkit_ , as I had expected after installing 'topydo[prompt]'.  I also tried without the --upgrade parameter.  At this point, is it supposed to be there?

As well, when I test-run the _Prompt mode_, an error message tells me the installation of dependencies failed, as follows:  

```
ron@ron-gnulinux:~/.local/bin > ./topydo prompt
Some additional dependencies for prompt mode were not installed, please install with 'pip install topydo[prompt]'
```

- I'm confused because this message is advising me to do a 'pip install' -- but not a 'pip3 install'  -- and I thought I had to avoid 'pip installs' because that was 'pulling in' incorrect, earlier versions.
- I'm also confused because the 'pip3 install' was already done, seems to have gone to completion and yet, a dependency is still missing -- and so, the _Prompt mode_ fails to start. 

P.S.: I now have PATH fixed, but still the same as above -- except I can just type _topydo_ by itself.

Any help will be most appreciated!

**Comment by mruwek** [comment]
If you're on Bash shell, try to remove those single quotes around `topydo[prompt]`. Also try to place `--` opts before, so the command becomes:

```
pip3 install --user --upgrade topydo[prompt]
```

> and I thought I had to avoid 'pip installs' because that was 'pulling in' incorrect, earlier versions.

That is correct. This message from topydo and quoted by you apparently needs to be corrected.

**Comment by RonCam** [comment]
Thanks again!
I did a copy-paste of your corrected command into terminal, with the following result:
```
ron@ron-gnulinux:~/.local/bin > pip3 install --user --upgrade topydo[prompt]
Requirement already up-to-date: topydo[prompt] in /home/ron/.local/lib/python3.4/site-packages
  Installing extra requirements: 'prompt'
Cleaning up...
ron@ron-gnulinux:~/.local/bin > topydo prompt
Some additional dependencies for prompt mode were not installed, please install with 'pip install topydo[prompt]'
```

- Still the same -- what else could be happening, that I may correct?


My PATH has been fixed and _topydo_ followed by parameters now works from anywhere on the system.

**Comment by mruwek** [comment]
Can you give us also output of `pip3 --version`?

**Comment by RonCam** [comment]
> Can you give us also output of pip3 --version?
```
ron@ron-gnulinux:~/.local/bin > pip3 --version
pip 1.5.4 from /usr/lib/python3/dist-packages (python 3.4)
```
Hope this helps ...

**Comment by mruwek** [comment]
Ok, it's a bug in your version of pip:  pypa/pip#1885. It was fixed in pip 6.0 (current version is 9.0.1 for the record, and yours is from 2014-02-21).

2 solutions:

1. Upgrade pip
a) system-wide: `sudo -H pip3 install --upgrade pip`
b) OR locally: `pip3 install --upgrade --user pip` - and create an alias for pip3 that uses your local version of pip: `alias pip3=~/.local/bin/pip3`
2. Don't use wheel installation: `pip3 install prompt_toolkit`
3. Reinstall: `pip3 uninstall topydo && pip3 install --user --upgrade topydo[columns,prompt]`

**Comment by RonCam** [comment]
That's great, I'm glad it wasn't 'just me'.
Hmmm ... I don't want to foul anything up.  Let me repeat this and please confirmthat  I'm understanding:

- The '2 solutions statement' refers _only_ to the upgrade methods, i.e., a) OR b) 
and, has nothing to do with the three lines numbered 1, 2 and 3.
- For step 2, I'm to input the terminal command you gave.  Is it correct, running that command has nothing to do with starting a 'wheel installation'?  Just checking, since that term is totally unfamiliar.  
- On the a) or b) upgrade choice, _for a single-user machine,_ may I assume that it would be the best and simplest choice -- no special aliases needed.

Thanks for your patience!  I suspect the answer will be 'yes for all three' but just wanted to verify.

**Comment by mruwek** [comment]
I've edited the reply already, so there is now "3 solutions" - I've made solution number 3 after writing that line :) Each of the solution should work.

> Is it correct, running that command has nothing to do with starting a 'wheel installation'?

topydo uses [wheel type of distribution](https://wheel.readthedocs.io/en/latest/), all you have to know is that this type of distribution is buggy in your version of pip.

> may I assume that it would be the best and simplest choice -- no special aliases needed

I don't know how your distro will behave if you'll update pip system-wide. 90% it will be just fine, but there is still some tiny risk, that you will fall into weird permissions related problems.

**Comment by RonCam** [comment]
Thanks for the extra explanation.
I decided, then, not to upgrade _pip_ and went instead with Solution 2, to produce the following result:
```
ron@ron-gnulinux:~/.local/bin > topydo prompt
topydo> 
```
Perfect!  I now have a _topydo prompt_, and _columns_ is (still!) working.  Now, to use _topydo_.
The _pip_ upgrade will most likely be taken care of, when I move from Linux Mint 17.3 to 18.

Your knowledge of Python is amazing!  I've learned a lot from reading your replies.
